### ISSUE TITLE
bug: cli args parsing fails

### ISSUE THREAD

**Issue by jasperbosch** [ionic-team/stencil#3481]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.x

### Current Behavior

When executing `stencil test --spec --e2e --json --outputFile=output.json` the outputFile-argument is badly parsed. It will result in `jest --spec --e2e --json --outputFile=output.json output.json`, wich will result in not executing jest.

However executing `stencil test --spec --e2e --json --output-file output.json` will succeed.

Cause is in the `parse-flags.ts` file:
```typescript
args.forEach((arg, i) => {
    if (arg.startsWith(`--${dashCaseName}=`) || arg.startsWith(`--${configCaseName}=`)) {
      value = getEqualsValue(arg);
      matchingArg = arg;
    } else if (arg === `--${dashCaseName}` || arg === `--${configCaseName}`) {
      value = args[i + 1];
      matchingArg = arg;
    } else if (alias) {
      if (arg.startsWith(`-${alias}=`)) {
        value = getEqualsValue(arg);
        matchingArg = arg;
      } else if (arg === `-${alias}`) {
        value = args[i + 1];
        matchingArg = arg;
      }
    }
  });
  ```
  `matchingArg` (on line 4) should only contain `--outputFile=` but now contains `--outputFile=output.json`

### Expected Behavior

Expect to execute jest with: `jest --spec --e2e --json --outputFile=output.json`

### Steps to Reproduce

execute `stencil test --spec --e2e --json --outputFile=output.json`

### Code Reproduction URL

-

**Comment by alicewriteswrongs** [comment]
hey @jasperbosch, thanks for filing this issue and for doing some code spelunking! You're right that this is an issue. I was just in this code recently so I'll fix it today.

**Comment by alicewriteswrongs** [comment]
Thanks for reporting this! Just to update you, there's a bugfix PR up here: https://github.com/ionic-team/stencil/pull/3483 which just merged into `main`.

If you want to test it out to make sure it fixes your issue you could

- clone the repo locally
- check out [the branch](https://github.com/ionic-team/stencil/tree/ap/flag-parsing-bug)
- do `npm install`, `npm run build` and `npm pack`
- try installing the resulting `.tgz` file in our project by doing `npm install ./path/to/stencil.tgz`

If you'd rather wait for a pre-release or release build to try that out no worries :)

**Comment by rwaskiewicz** [comment]
The fix for this issue has been officially released as a part of [Stencil v2.17.2](https://github.com/ionic-team/stencil/releases/tag/v2.17.2). As a result, I'm going to close this issue and mark it as completed. If the bug re-appears, please feel free to open a new issue!
